sdk/core/azure-core/src/main/java/com/azure/core/http/HttpHeaders.java

```diff
@@ -40,11 +42,21 @@ public HttpHeaders(Map<String, String> headers) {
      * @param headers the collection of initial headers
      */
     public HttpHeaders(Iterable<HttpHeader> headers) {
+        this.headers = new HashMap<>();
         for (final HttpHeader header : headers) {
             this.set(header.getName(), header.getValue());
         }
     }
 
+    /**
+     * Create a HttpHeaders instance with an initial {@code size} empty headers
+     *
+     * @param initialCapacity the initial capacity of headers map.
+     */
+    public HttpHeaders(int initialCapacity) {
+        this.headers = new HashMap<>(initialCapacity);
```

Comment:
Should this validate `initialCapacity`?

Reply:
Do you mean we should validate the initialCapacity to be less than 2^30, which is the max capacity of Java hashmap.

Reply:
Actually, let's ignore this and let HashMap handle throwing the error if the `initialCapicity` is invalid